drivers/flash/spi_nor.c

```diff
@@ -1246,6 +1254,44 @@ static int spi_nor_configure(const struct device *dev)
 	return 0;
 }
 
+#ifdef CONFIG_PM_DEVICE
+
+static int spi_nor_pm_control(const struct device *dev, enum pm_device_action action)
```

Comment:
There is not check whether device is busy at the moment.
Look at `acquire_device` and `release_device`, which are called when device is accessed and released respectively: they use `spi_nor_data->sem` to prevent DPD when in multithreading. Shouldn't this also happen here?